app/main.py

```diff
@@ -1 +1,41 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+    alive: list[Animal] = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        if self.health > 0:
+            Animal.alive.append(self)
+
+    def die(self) -> None:
+        if self.health <= 0:
+            if self in Animal.alive:
+                Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, target: Animal) -> None:
+        if isinstance(target, Herbivore) and not target.hidden:
+            target.health -= 50
+            if target.health <= 0:
+                target.die()
```

Comment:
You check if health is less than 0 in 2 places, maybe you should remove this check from somewhere?